packages/desktop-client/src/components/reports/CategorySelector.tsx

```diff
@@ -24,23 +26,72 @@ export default function CategorySelector({
 }: CategorySelectorProps) {
   const [uncheckedHidden, setUncheckedHidden] = useState(false);
 
+  const allCategoriesSelected = categoryGroups.every(categoryGroup =>
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏The amount of loops within loops within loops here is crazy. There's definitely things we could do to bring this down from O(n^3). Possibly building a hashmap of the available categories one time and then just using that to compare against `selectedCategories`.

Also: should wrap this in `useMemo`.

Reply:
I've never done a hashmap. Any help would be appreciated.

Reply:
I've imported categories.list. Is this better? It's only removed one loop...Thoughts?

categories.every(category => 
    selectedCategories.map(selected =>
      selected.id).includes(category.id));